cli/azd/pkg/azapi/standard_deployments.go

```diff
@@ -691,3 +694,108 @@ func convertFromStandardProvisioningState(state armresources.ProvisioningState)
 
 	return DeploymentProvisioningState("")
 }
+
+func (ds *StandardDeployments) ValidatePreflightToSubscription(
+	ctx context.Context,
+	subscriptionId string,
+	location string,
+	deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+	tags map[string]*string,
+	options map[string]any,
+) error {
+	deploymentClient, err := ds.createDeploymentsClient(ctx, subscriptionId)
+	if err != nil {
+		return fmt.Errorf("creating deployments client: %w", err)
+	}
+
+	var rawResponse *http.Response
+	ctxWithResp := policy.WithCaptureResponse(ctx, &rawResponse)
+
+	validateResult, err := deploymentClient.BeginValidateAtSubscriptionScope(
+		ctxWithResp, deploymentName,
+		armresources.Deployment{
+			Properties: &armresources.DeploymentProperties{
+				Template:   armTemplate,
+				Parameters: parameters,
+				Mode:       to.Ptr(armresources.DeploymentModeIncremental),
+			},
+			Location: to.Ptr(location),
+			Tags:     tags,
+		}, nil)
+	if err != nil {
+		return validatePreflightError(rawResponse, err, "subscription")
+	}
+	_, err = validateResult.PollUntilDone(ctx, nil)
+	if err != nil {
+		return validatePreflightError(rawResponse, err, "subscription")
+	}
+
+	return nil
+}
+
+func validatePreflightError(
+	rawResponse *http.Response,
+	err error,
+	typeMessage string,
+) error {
+	var respErr *azcore.ResponseError
+	if errors.As(err, &respErr) {
+		err = createDeploymentError(err)
+	} else if err != nil {
+		// Error returned from azure sdk go bug
+		// https://github.com/Azure/azure-sdk-for-go/issues/23350
+		if rawResponse == nil || rawResponse.StatusCode != 400 {
```

Comment:
I suspect that this if-condition may need to be tweaked slightly from the previous iteration -- here we're trying to execute the if-block only if we detect that a 400 response is returned

```suggestion
		if rawResponse != nil && rawResponse.StatusCode == 400 {
```